frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx

```diff
@@ -347,6 +353,8 @@ export default class NativeQueryEditor extends Component {
           .update(this.props.setDatasetQuery);
       }
     }
+
+    this._retriggerAutocomplete();
```

Comment:
The drawback of this approach: potential flicker in the UI, though it's likely more observed only in certain cases.

Reply:
I wonder if we could fix the flicker issue, as sometimes it would look weird. See the video below where I throttle the network speed to see how the flicker occurs. And for anything input that would already triggers `getCompletions` e.g. `p` in this case, will call the API to `/api/database/1/autocomplete_suggestions?prefix=p` twice.
1. When `getCompletions` is called the first time
2. When `_retriggerAutocomplete` is called (700ms later)

I'm also not sure what caused the result to change while we're waiting to get a new result from `this.props.autocompleteResultsFn(prefix)` as you can see from the video below when I type more characters the autocomplete result changed then after API finished loading the result changed again.

https://user-images.githubusercontent.com/1937582/158120281-638dcb14-f70d-497a-8019-fdb21909eaee.mov

Reply:
> I'm also not sure what caused the result to change while we're waiting to get a new result from `this.props.autocompleteResultsFn(prefix)` as you can see from the video below when I type more characters the autocomplete result changed then after API finished loading the result changed again.

I believe the first suggested matches from `p`, i.e. `primary`, `update`, `type`, `drop`, `group`, `timestamp` are all parts of Ace's basic completion. Note how all of them are just SQL keywords (this editor is set to use SQL mode) which contains the letter `p`.

After a while, the `/api/database/1/autocomplete_suggestions?prefix=p` returns some results and now Ace populates the suggestion with extra matches, e.g. `PASSWORD`, `PEOPLE`, `PRICE`, etc.

The same cycle happens again when the user types more characters.

Reply:
That makes sense. I wish there was a way to persist autocomplete results when typing more characters though. e.g. after `p` autocomplete results were populated, typing another `a` shouldn't disregard the previous results and fall back to using Ace's default autocomplete.